### PR TITLE
Danroscigno container docs

### PR DESCRIPTION
**Pull request by DanRoscigno**
These are updates to the docs and to the files metricbeat/metricbeat.docker.yml and filebeat/filebeat.docker.yml.  The updates consist of:

 - Replacing the docker pull command on the Install pages with links to the Running on Docker pages
 - Adding required Linux Capabilities to the commands
 - Adding setup commands
 - Adding hints based Autodiscover to the sample metricbeat.docker.yml and filebeat.docker.yml files

### CODE REVIEW

File: deploy/docker/filebeat.docker.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+filebeat.modules:
```
Comment (dedemorton) [inline deploy/docker/filebeat.docker.yml]: make this change in `_meta/beat.docker.yml` and run `make update`

File: filebeat/filebeat.docker.yml [hunk 2]
```diff
@@ -3,6 +3,12 @@ filebeat.config:
     path: ${path.config}/modules.d/*.yml
     reload.enabled: false
 
+# To enable hints based autodiscover, remove `filebeat.config.inputs` configuration and uncomment this:
```
Comment (dedemorton) [inline filebeat/filebeat.docker.yml]: Make this change in `/_meta/beat.docker.yml1

File: metricbeat/metricbeat.docker.yml [hunk 3]
```diff
@@ -1,9 +1,32 @@
-metricbeat.config.modules:
-  path: ${path.config}/modules.d/*.yml
-  reload.enabled: false
+metricbeat.config:
```
Comment (dedemorton) [inline metricbeat/metricbeat.docker.yml]: Make this change in `/_meta/beat.docker.yml`

File: filebeat/filebeat.docker.yml [hunk 4]
```diff
@@ -3,6 +3,12 @@ filebeat.config:
     path: ${path.config}/modules.d/*.yml
```
Comment (exekias) [inline filebeat/filebeat.docker.yml]: ey @DanRoscigno, this is not a sample file, this is the default config we ship with our containers (starting with 7.0), is this change intended?

File: libbeat/docs/shared-docker.asciidoc [hunk 5]
```diff
@@ -41,26 +41,183 @@ https://www.docker.elastic.co[www.docker.elastic.co].
...
-===== Bind-mounted configuration
+===== The configuration file
 
-One way to configure {beatname_uc} on Docker is to provide +{beatname_lc}.yml+ via a bind mount.
-With +docker run+, the bind mount can be specified like this:
+Download this file as a starting point:
+
+["source","sh",subs="attributes,callouts"]
+------------------------------------------------
+curl -L -O https://raw.githubusercontent.com/elastic/beats/{doc-branch}/{beatname_lc}/{beatname_lc}.docker.yml
```
Comment (exekias) [inline libbeat/docs/shared-docker.asciidoc]: I think you want to use `deploy/docker/{beatname_lc}...`?

File: filebeat/filebeat.docker.yml (hunk 4 above)
Comment (DanRoscigno) [reply filebeat/filebeat.docker.yml]: I think that it would be great to have a default file that causes some data to be sent with no configuration, so if we should always get data flowing if the system module is included then I would do that.  But, I did not know that filebeat.docker.yml would be defaulted to in 7, that is great.  I did not intend to change the path if that is what you are asking.  I will take a look in detail tonight, I am in all day meetings and sneaking on to quickly read comments :)

**Comment by DanRoscigno** [comment]
Resubmitting with requested changes.

**Comment by dedemorton** [comment]
Replaced with https://github.com/elastic/beats/pull/9391
